src/Hyperion/SerializerFactories/EnumerableSerializerFactory.cs

```diff
@@ -83,6 +83,7 @@ private static Type GetEnumerableType(Type type)
         private static ConstructorInfo GetEnumerableConstructor(Type type)
         {
             var enumerableType = GetEnumerableType(type);
+            enumerableType = enumerableType ?? typeof(object);
```

Comment:
What's this for?

Reply:
GetEnumerableType is made for generics and works great for them, however, for non-generics that store objects, it just returns null, hence the check and by default the usage of typeof(object).

Reply:
@Aaronontheweb hello ?

Reply:
Sorry for the delay - missed the Github notification on this (I get a lot of them.)

Seems like we should make this check more robust though - some pseudo-code:

```
if type.IsGeneric
   return GetEnumerableType(type)
else
  return typeof(object)
```

At this point in the serializer's execution, do we know already that we're dealing with an `IEnumerable` type (I assume we are)? If the answer is "yes," then at the very least we should add some comments here that explain why this check is in-place. Otherwise it'd be easy to miss.